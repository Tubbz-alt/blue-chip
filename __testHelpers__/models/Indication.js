import BaseModel from "../../src/BaseModel";
import Patient from "./Patient";

export default class Indication extends BaseModel {
  static get hasMany() {
    return [Patient];
  }
}
