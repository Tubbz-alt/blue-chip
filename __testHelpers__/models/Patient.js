import BaseModel from "../../src/BaseModel";
import SettingOfUse from "./SettingOfUse";
import Indication from "./Indication";

export default class Patient extends BaseModel {
  static get belongsTo() {
    return [SettingOfUse, Indication];
  }
}
