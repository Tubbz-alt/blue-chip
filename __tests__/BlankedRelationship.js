import resources from "../__testHelpers__/fixtures/patientsAndSettingsOfUseNormalized";
import Patient from "../__testHelpers__/models/Patient";
import Indication from "../__testHelpers__/models/Indication";

describe("Blank relationship", () => {
  test("One relationship blank should still return the non blank relationships", () => {
    const resourcesClone = JSON.parse(JSON.stringify(resources))
    resourcesClone.patients["667"].relationships.settingOfUse.data = null

    const patients = Patient.query(resourcesClone)
      .includes(["indication", "settingOfUse"])
      .toObjects();
    expect(patients).toMatchSnapshot();
  });
});
