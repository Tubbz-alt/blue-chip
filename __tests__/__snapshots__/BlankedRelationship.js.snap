// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blank relationship One relationship blank should still return the non blank relationships 1`] = `
Array [
  Object {
    "createdAt": "2018-10-08T18:39:33.000Z",
    "customers": Array [
      "0000",
    ],
    "id": "667",
    "indication": Object {
      "id": "1",
      "name": "SCI",
    },
    "lastSessionDate": "2016-10-01T00:00:00.000Z",
    "myPatient": true,
    "patientId": "0000",
    "serialNumbers": Array [
      "21601009",
    ],
    "sessionsCount": 127001,
    "steps": 24099980,
    "upTime": 68599053,
  },
  Object {
    "createdAt": "2018-10-08T18:39:33.000Z",
    "customers": Array [
      "0001",
    ],
    "id": "668",
    "indication": Object {
      "id": "2",
      "name": "Stroke",
    },
    "lastSessionDate": "2016-10-01T00:00:00.000Z",
    "myPatient": true,
    "patientId": "0001",
    "serialNumbers": Array [
      "21601009",
    ],
    "sessionsCount": 127001,
    "settingOfUse": Object {
      "id": "2",
      "name": "Patient",
    },
    "steps": 24099980,
    "upTime": 68599053,
  },
  Object {
    "createdAt": "2018-10-08T18:39:33.000Z",
    "customers": Array [
      "0002",
    ],
    "id": "669",
    "indication": Object {
      "id": "3",
      "name": "Something Else",
    },
    "lastSessionDate": "2016-10-01T00:00:00.000Z",
    "myPatient": true,
    "patientId": "0002",
    "serialNumbers": Array [
      "21601009",
    ],
    "sessionsCount": 127001,
    "settingOfUse": Object {
      "id": "1",
      "name": "Inpatient",
    },
    "steps": 24099980,
    "upTime": 68599053,
  },
]
`;
